prow/config/jobs.go

```diff
@@ -190,6 +191,13 @@ type Brancher struct {
 	reSkip *regexp.Regexp
 }
 
+// ChangeMatcher is for code shared between jobs that run only when certain files are changed.
+type ChangeMatcher struct {
+	// RunIfChanged automatically run if the PR modifies a file that matches this regex.
+	RunIfChanged string         `json:"run_if_changed,omitempty"`
+	reChanges    *regexp.Regexp // from RunIfChanged
```

Comment:
This name will be unwieldy if there's ever another type of `ChangeMatcher` backing impl in this struct